ml_frameworks_bot/keras/docs_agent.py

```diff
@@ -19,21 +24,21 @@ class KerasOpWithAPIReference(BaseModel):
 class KerasDocumentationAgent(weave.Model):
     op_extraction_llm_client: LLMClientWrapper
     retrieval_augmentation_llm_client: LLMClientWrapper
```

Comment:
We probably can skip the `ask_llm_about_op` operation using the `retrieval_augmentation_llm_client` when using the `KerasDocumentationHeuristicRetreiver`, right?

Reply:
```python
is_neural_retriever = isinstance(
    self.api_reference_retriever, 
    KerasDocumentationRetreiver
)
if is_neural_retriever:
    purpose_of_op = self.ask_llm_about_op(keras_op)
    api_reference: BaseNode = self.api_reference_retriever.predict(
        query=f"API reference for `{keras_op}`.\n{purpose_of_op}",
    )[0]
else:
    api_reference: BaseNode = self.api_reference_retriever.predict(
        query=keras_op
    )
```

Yes, we skip the `ask_llm_about_op` call when using the heuristic retriever.